pyiceberg/table/metadata.py

```diff
@@ -411,6 +442,24 @@ def new_table_metadata(
     if table_uuid is None:
         table_uuid = uuid.uuid4()
 
+    # Remove format-version so it does not get persisted
+    format_version = int(properties.pop("format-version", DEFAULT_FORMAT_VERSION))
```

Comment:
What do you think of moving the literal and constant to `TableProperties`?

Reply:
We can and probably should but in this PR we can't reference it in `metadata.py` since it ends up being a circular reference. 

```
 ImportError: cannot import name 'TableProperties' from partially initialized module 'pyiceberg.table' (most likely due to a circular import) (/home/amogh/workspace/iceberg-python/pyiceberg/table/__init__.py
```

I think the way to handle that is by moving it to a separate file but still ensure that we preserve backwards compatibility for any code which was importing TableProperties from the original table namespace.

I think for now we can just have the local constants and have an issue to handle the organization of TableProperties in a future PR? Unless I'm missing a simple way to address this right here.

Reply:
Actually, I think the error message is a bit misleading it can be addressed by localizing the import in the function where it's used.